Tests/Functional/GraphQL/AssetApiTest.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flowpack\Media\Ui\Tests\Functional\GraphQL;
+
+/*
+ * This file is part of the Flowpack.Media.Ui package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Flowpack\Media\Ui\GraphQL\MediaApi;
+use Flowpack\Media\Ui\GraphQL\Resolver\Type\AssetResolver;
+use Flowpack\Media\Ui\GraphQL\Types;
+use Flowpack\Media\Ui\Tests\Functional\AbstractMediaTestCase;
+use Neos\Flow\Persistence\Doctrine\PersistenceManager;
+use Neos\Flow\Tests\Behavior\Features\Bootstrap\SecurityOperationsTrait;
+use Neos\Utility\Files;
+
+use function Wwwision\Types\instantiate;
+
+/**
+ * Testcase for the Media.Ui API
+ */
+class AssetApiTest extends AbstractMediaTestCase
+{
+    use SecurityOperationsTrait;
+
+    /**
+     * @var boolean
+     */
+    protected static $testablePersistenceEnabled = true;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->isolated = false;
+        if (!$this->persistenceManager instanceof PersistenceManager) {
+            static::markTestSkipped('Doctrine persistence is not enabled');
+        }
+
+        $this->mediaApi = $this->objectManager->get(MediaApi::class);
+        $this->assetResolver = $this->objectManager->get(AssetResolver::class);
+
+        $this->iAmAuthenticatedWithRole('Neos.Neos:Editor');
+    }
+
+    protected static function createFile(): Types\UploadedFile
+    {
+        $fileContent = Files::getFileContents(__DIR__ . '/Fixtures/norman.svg');
+        return instantiate(Types\UploadedFile::class, [
+            'streamOrFile' => $fileContent,
+            'size' => strlen($fileContent),
+            'clientMediaType' => 'image/svg+xml',
+            'clientFilename' => 'test.svg',
+            'errorStatus' => 0,
+        ]);
+    }
+
+    public function testUploadFile(): void
+    {
+        $file = self::createFile();
+
+        $result = $this->mediaApi->uploadFiles(
+            Types\UploadedFiles::fromArray([$file])
+        );
+
+        $this->assertCount(1, $result->values);
+        $uploadResult = $result->getIterator()->current();
+        $this->assertInstanceOf(Types\FileUploadResult::class, $uploadResult);
+        $this->assertTrue($uploadResult->success);
+        $this->assertEquals('test.svg', $uploadResult->filename->value);
+    }
+
+    public function testEditAsset(): void
```

Comment:
thanks for the tests ❤️  also cool to see the php api with the types being used directly that is FUN :D